### DISCUSSION TITLE
Do you scan emails before forwarding?

### DISCUSSION THREAD

**Discussion by merrpink** [anonaddy/anonaddy#247] [General]
This is an important privacy concern of mine. Can someone please tell me if anonaddy scans emails so I don't have to go through all the code. Thanks.

**Comment by ccorneli** [comment]
> This is an important privacy concern of mine. Can someone please tell me if anonaddy scans emails so I don't have to go through all the code. Thanks.


Did you read the FAQ?
https://anonaddy.com/faq/#do-you-store-emails
There are quite a few security/privacy related questions there. 

Or do you have concerns beyond that?  (If so, Please elaborate)

**Reply by merrpink** [reply]
Thanks for your reply ccorneli. Yes I did read the FAQ. In fact I read almost the whole site very carefully and I liked what I saw. My concern is not whether mails are stored or not, but whether they are scanned or not (as I asked in my original post). For example I discovered that simplelogin scans all mails (for spam supposedly). Obviously, if scanning is going on, that opens the doors for all kinds of abuses. So please tell me if anonaddy believes that scanning is necessary to "protect" its customers or not. Obviously I'm referring to non-encrypted mails by the way. Thanks.

**Comment by ccorneli** [comment]
I saw someone recently [posted a question here](https://github.com/anonaddy/anonaddy/issues/246) email being blocked from some particular domain, I don't think it has been answered yet.  But I'm inclined to think it is some issue with the sender, not active filtering by AnonAddy..   
I don't think it does any form of 'protection' in that sense, so I'm curious on the response. 

But of course it has the capability to do so, since it needs to scan and rewrite both the header and message body for normal processing.  (But I don't think your concern is having access, only what it does with it...)

For example, I was affected by the ['inline-image' issue](https://github.com/anonaddy/anonaddy/issues/210), and took a brief look at the code fix here: [Added support for inline images](https://github.com/anonaddy/anonaddy/commit/1ea9ddefae6cd51bf9bcf45ffdce71414d6b1d5f)

And if I'm reading it right, you can see where it stores the message body:
```
$this->emailText = $emailData->text;
$this->emailHtml = $emailData->html;
```
But again, I'm not aware of any processing, filtering that is done.

Also, as for AnonAddy automatically Encrypting email... It would still start with the non-encrypted copy, so if it is happening at all, it would be a concern either way.

**Reply by merrpink** [reply]
You make some good points. The best way is for the creator to clarify exactly what is happening here. Nullbox, for example, explicitly says no scanning, and simplelogin admits to scanning (although only for 'protection'). Anonaddy should clarify I think.

**Comment by ccorneli** [comment]
Agreed.. hopefully @willbrowningme can elaborate in the FAQ to include more of a description of explicitly what kind of "filtering, processing, etc.." is done (or not done) as required.   Since just saying 'save/store' leaves a bit of wiggle room for whatever happens in transit as it is processed.    Somewhat related: I remember seeing a feature request to allow user-defined "rule processing" on the messages as they are received.  I don't remember the outcome, but I think for now it was considered not really in scope for AA and better supported in a mail client which saves your email... but it is something to consider mentioning if it is on the future roadmap.

**Comment by willbrowningme** [comment]
[This question](https://anonaddy.com/faq/#how-do-you-prevent-spammers) in the FAQ states that [Rspamd](https://rspamd.com/) is used on the server which is an open-source spam filtering system.

The main reason this software is on the server is to help protect the mail server's sending reputation. If the AnonAddy mail server was frequently forwarding/sending spam/junk/phishing/viruses etc. then its reputation could easily be damaged.

You can find more details about the Rspamd setup in the [self-hosting](https://anonaddy.com/self-hosting/) instructions.

**Reply by beedell-roke** [reply]
I'd mark this as the solution. Thanks.
